src/app/models/tag.ts

```diff
@@ -0,0 +1,11 @@
+import { Card } from "src/app/models/card";
+import { Stack } from "src/app/models/stack";
+
+export class Tag{
+    constructor(
+        public tagId:number,
+        public tagName:string,
+        public stacks:Stack[],
+        public cards:Card[]
```

Comment:
Please remove stacks and cards, the relationship will only be accessible through the stacks and cards and not the tags.